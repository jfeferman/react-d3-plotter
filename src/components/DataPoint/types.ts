export type DataPointProps = {
  coords: Array<number>,
  xScale: Function,
  yScale: Function,
}
