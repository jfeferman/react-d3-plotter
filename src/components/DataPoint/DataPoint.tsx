type DataPointProps = {
  coords: number[]
  xScale: () => number
  yScale: () => number
  radius: number
}

export const DataPoint = ({
  coords, xScale, yScale, radius = 2,
}: DataPointProps) => {
  return (
    <circle
      cx={xScale(coords[0])}
      cy={yScale(coords[1])}
      r={radius}
      key={Math.random() * 1}
    />
  )
}
