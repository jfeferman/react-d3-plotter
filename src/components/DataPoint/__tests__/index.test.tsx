import React from 'react'
import * as TestRenderer from 'react-test-renderer';
import {
  mockData,
  getXScale,
  getYScale,
} from '../../../utils/testUtil'
import DataPoint from '..'

describe('compponents/DataPoint', () => {
  let renderer: TestRenderer.ReactTestRenderer;

  beforeAll(() => {
    renderer = TestRenderer.create(
      <DataPoint
        coords={mockData[0]}
        xScale={getXScale()}
        yScale={getYScale()}
      />,
    )
  })

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
