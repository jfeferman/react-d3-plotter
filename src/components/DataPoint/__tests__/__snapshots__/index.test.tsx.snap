// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compponents/DataPoint matches the snapshot of the component 1`] = `
<circle
  cx={480}
  cy={60}
/>
`;
